yarn-project/sequencer-client/src/sequencer/sequencer.ts

```diff
@@ -255,72 +255,72 @@ export class Sequencer {
     if (!chainTip) {
       return;
     }
+    const chainTipArchive = chainTip.archive;
+    const newBlockNumber = chainTip.blockNumber + 1;
 
     this.setState(SequencerState.PROPOSER_CHECK, 0n);
 
-    const newBlockNumber = chainTip.blockNumber + 1;
-
-    // If we cannot find a tip archive, assume genesis.
-    const chainTipArchive = chainTip.archive;
+    const result = await this.publisher.canProposeAtNextEthBlock();
 
-    const slot = await this.slotForProposal(chainTipArchive.toBuffer(), BigInt(newBlockNumber));
-    if (!slot) {
-      this.log.debug(`Cannot propose block ${newBlockNumber}`);
+    if (!result) {
+      this.log.debug(`Not proposer`);
       return;
     }
 
-    this.log.debug(`Can propose block ${newBlockNumber} at slot ${slot}`);
+    const [slot, slotTimestamp, blockNumber, tipArchive] = result;
+    this.log.debug(`Is proposer for slot ${slot}`);
 
-    const newGlobalVariables = await this.globalsBuilder.buildGlobalVariables(
-      new Fr(newBlockNumber),
-      this._coinbase,
-      this._feeRecipient,
-      slot,
-    );
+    const enqueueGovernanceVotePromise = this.publisher.enqueueCastVote(slot, slotTimestamp, VoteType.GOVERNANCE);
+    const enqueueSlashingVotePromise = this.publisher.enqueueCastVote(slot, slotTimestamp, VoteType.SLASHING);
 
-    const enqueueGovernanceVotePromise = this.publisher.enqueueCastVote(
-      slot,
-      newGlobalVariables.timestamp.toBigInt(),
-      VoteType.GOVERNANCE,
-    );
-    const enqueueSlashingVotePromise = this.publisher.enqueueCastVote(
-      slot,
-      newGlobalVariables.timestamp.toBigInt(),
-      VoteType.SLASHING,
-    );
+    let finishedFlushing = false;
 
-    this.setState(SequencerState.INITIALIZING_PROPOSAL, slot);
-    this.log.verbose(`Preparing proposal for block ${newBlockNumber} at slot ${slot}`, {
-      chainTipArchive,
-      blockNumber: newBlockNumber,
-      slot,
-    });
+    // Only if we are caught up will we try to make a proposal
+    if (!chainTipArchive.equals(Fr.fromHexString(tipArchive)) || BigInt(newBlockNumber) !== blockNumber) {
```

Comment:
I think we need a test to make sure if we end up hitting this check, then sending a vote, that we can still make a proposal within our slot